### PR TITLE
   [EVENTVWR] Add a context menu to TreeView items. Same as WindowsXP

### PR DESCRIPTION
**Pull request by zecarlos1957**
[EVENTVWR] Add a context menu to TreeView items. Same as WindowsXP
                       Add predefined file extension when opening or saving a event log file.

Expected behavior:
If the user does not define an extension to the filename, ReactOs adds the default value.

### CODE REVIEW

File: base/applications/mscutils/eventvwr/eventvwr.c [hunk 1]
```diff
@@ -3807,6 +3807,36 @@ WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
             break;
         }
 
+        case WM_CONTEXTMENU:
```
Comment (HBelusca) [inline base/applications/mscutils/eventvwr/eventvwr.c:3810]: Please move this code to where there was an existing `WM_CONTEXTMENU` that was commented out.

File: base/applications/mscutils/eventvwr/eventvwr.c [hunk 2]
```diff
@@ -3807,6 +3807,36 @@ WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
+            hInfo.pt.y = yPos - rc.top;
+
+            if ((hItem = TreeView_HitTest(hwndTreeView, &hInfo)))
+            {
+                TreeView_SelectItem(hwndTreeView, hItem);
+
+                if (TreeView_GetParent(hwndTreeView, hItem))
+                {
+                    HMENU hCtxtMenuBar = LoadMenu(hInst, MAKEINTRESOURCE(IDM_EVENTWR_CTX));
+                    HMENU contextMenu = GetSubMenu(hCtxtMenuBar, 0);
```
Comment (HBelusca) [inline base/applications/mscutils/eventvwr/eventvwr.c:3830]:
```suggestion
                    HMENU hCtxMenu = GetSubMenu(LoadMenuW(hInst, MAKEINTRESOURCEW(IDM_EVENTWR_CTX)), 0);
```

File: base/applications/mscutils/eventvwr/eventvwr.c [hunk 3]
```diff
@@ -3807,6 +3807,36 @@ WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
+
+                    SendMessage(hWnd, WM_COMMAND, TrackPopupMenuEx(contextMenu,
+                                TPM_LEFTALIGN|TPM_TOPALIGN|TPM_NONOTIFY|TPM_RETURNCMD,
+                                xPos, yPos, hWnd, NULL), (LPARAM)hwndTreeView);
```
Comment (HBelusca) [inline base/applications/mscutils/eventvwr/eventvwr.c:3834]: Why isn't `TrackPopupMenuEx()` all by itself (without the SendMessage stuff etc.) sufficient?

File: base/applications/mscutils/eventvwr/eventvwr.c [hunk 4]
```diff
@@ -3985,7 +4015,7 @@ InitPropertiesDlg(HWND hDlg, PEVENTLOG EventLog)
 
     if (EventLog->Permanent)
     {
-        SendDlgItemMessageW(hDlg, IDC_UPDOWN_MAXLOGSIZE, UDM_SETRANGE32, (WPARAM)1, (LPARAM)0x3FFFC0);
+        SendDlgItemMessageW(hDlg, IDC_UPDOWN_MAXLOGSIZE, UDM_SETRANGE32, (WPARAM)64, (LPARAM)0x3FFFC0);
```
Comment (HBelusca) [inline base/applications/mscutils/eventvwr/eventvwr.c:4018]: When I coded that thing I wanted a finer granularity than 64kb jumps. So please revert. There is no point in adding artificial limitations.

File: base/applications/mscutils/eventvwr/eventvwr.c [hunk 5]
```diff
@@ -4107,6 +4137,18 @@ EventLogPropProc(HWND hDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
                     PropSheet_UnChanged(GetParent(hDlg), hDlg);
                     SavePropertiesDlg(hDlg, EventLog);
                     return (INT_PTR)TRUE;
+
+                 case UDN_DELTAPOS:
```
Comment (HBelusca) [inline base/applications/mscutils/eventvwr/eventvwr.c:4141]: See comment above.

**Review by HBelusca — CHANGES_REQUESTED** [review]

File: base/applications/mscutils/eventvwr/eventvwr.c [hunk 6]
```diff
@@ -3674,10 +3674,36 @@ WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
             lParam = lParam;
             break;
 
+#endif
```
Comment (HBelusca) [inline base/applications/mscutils/eventvwr/eventvwr.c:3677]:
```suggestion
#endif

```

File: base/applications/mscutils/eventvwr/lang/en-US.rc [hunk 7]
```diff
@@ -46,6 +46,19 @@ BEGIN
     END
 END
 
+IDM_EVENTWR_CTX MENU
+BEGIN
+    POPUP "popup"
```
Comment (HBelusca) [inline base/applications/mscutils/eventvwr/lang/en-US.rc:51]:
```suggestion
    POPUP ""
```
No need to use a title (just use empty string)

File: base/applications/mscutils/eventvwr/lang/en-US.rc [hunk 8]
```diff
@@ -48,7 +48,7 @@ END
 
 IDM_EVENTWR_CTX MENU
 BEGIN
-    POPUP "popup"
+    POPUP ""
```
Comment (HBelusca) [inline base/applications/mscutils/eventvwr/lang/en-US.rc:51]: Applies to all the other rc's as well.

Comment (zecarlos1957) [reply base/applications/mscutils/eventvwr/lang/en-US.rc:51]: YES!!! of course

File: base/applications/mscutils/eventvwr/eventvwr.c [hunk 9]
```diff
@@ -3673,11 +3673,37 @@ WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
+            TVHITTESTINFO hInfo = {0};
+
+            INT xPos = GET_X_LPARAM(lParam);
+            INT yPos = GET_Y_LPARAM(lParam);
+
+            GetWindowRect(hwndTreeView, &rc);
+            hInfo.pt.x = xPos - rc.left;
+            hInfo.pt.y = yPos - rc.top;
+
+            if ((hItem = TreeView_HitTest(hwndTreeView, &hInfo)))
```
Comment (HBelusca) [inline base/applications/mscutils/eventvwr/eventvwr.c:3691]:
```suggestion
            hItem = TreeView_HitTest(hwndTreeView, &hInfo);
            if (hItem)
```

File: base/applications/mscutils/eventvwr/eventvwr.c [hunk 10]
```diff
@@ -3673,11 +3673,37 @@ WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
+                TreeView_SelectItem(hwndTreeView, hItem);
+
+                if (TreeView_GetParent(hwndTreeView, hItem))
+                {
+                    HMENU hCtxMenu = GetSubMenu(LoadMenuW(hInst, MAKEINTRESOURCEW(IDM_EVENTWR_CTX)), 0);
+
+                    DWORD dwCmdID = TrackPopupMenuEx(hCtxMenu,
+                                                     TPM_LEFTALIGN | TPM_TOPALIGN | TPM_NONOTIFY | TPM_RETURNCMD,
+                                                     xPos, yPos, hWnd, NULL);
+                    SendMessage(hWnd, WM_COMMAND, (WPARAM)dwCmdID, (LPARAM)hwndTreeView);
```
Comment (HBelusca) [inline base/applications/mscutils/eventvwr/eventvwr.c:3702]:
```suggestion
                    SendMessageW(hWnd, WM_COMMAND, (WPARAM)dwCmdID, (LPARAM)hwndTreeView);
```

**Review by HBelusca — COMMENTED** [review]
Almost there 😂

File: base/applications/mscutils/eventvwr/eventvwr.c (hunk 9 above)
Comment (zecarlos1957) [reply base/applications/mscutils/eventvwr/eventvwr.c]: It's not the first time you've shown me this change. 
Hope next time I remember this

**Review by HBelusca — APPROVED** [review]

File: base/applications/mscutils/eventvwr/eventvwr.c (hunk 9 above)
Comment (HBelusca) [reply base/applications/mscutils/eventvwr/eventvwr.c]: That's about avoiding assignments within if or while conditions 😉

**Pull request merged**